app/models/manageiq/providers/kubevirt/inventory/parser.rb

```diff
@@ -164,14 +164,24 @@ def process_domain(memory, cores, uid, name)
 
     # Create the inventory object for the hardware:
     hw_object = hw_collection.find_or_build(vm_object)
-    hw_object.memory_mb = ManageIQ::Providers::Kubevirt::MemoryCalculator.convert(memory, 'Mi')
+    hw_object.memory_mb = parse_quantity(memory) / 1.megabytes.to_f
     hw_object.cpu_cores_per_socket = cores
     hw_object.cpu_total_cores = cores
 
     # Return the created inventory object:
     vm_object
   end
 
+  def parse_quantity(value)
+    return nil if value.nil?
+
+    begin
+      value.iec_60027_2_to_i
+    rescue
```

Comment:
Should we just rescue ArgumentError only here?  Not sure.

Reply:
So I pulled this from the kubernetes parser, figure we can refactor to pull that out into a module in the future and this is the way it handles the rescue

Reply:
+1 for moving this out into a shared module, then this provider can reuse it from the kube provider.